cloud/scope/vpc_cluster.go

```diff
@@ -221,6 +222,89 @@ func (s *VPCClusterScope) CheckTagExists(tagName string) (bool, error) {
+	return subnets, fmt.Errorf("error no control plane subnets available in status")
+}
+
+// GetLoadBalancerHostName will return the hostname of the cluster's public Load Balancer, assuming only one public Load Balancer was provided. Or, the hostname of the single private Load Balancer (assuming the cluster has no public access and only one private Load Balancer was provided).
+// NOTE(cjschaef): A webhook validation check could help ensure this.
+func (s *VPCClusterScope) GetLoadBalancerHostName() (*string, error) {
+	// If no Status or Load Balancer Status is populated, assume the Load Balancer's are not ready (have not been reconciled), so no hostname will be available.
+	if s.NetworkStatus() == nil || s.NetworkStatus().LoadBalancers == nil || len(s.NetworkStatus().LoadBalancers) == 0 {
+		return nil, nil
+	}
+
+	// If no Load Balancer's were defined, return an error, as a Load Balancer must be defined (no default Load Balancer is supported currently).
+	if len(s.NetworkSpec().LoadBalancers) == 0 {
+		return nil, fmt.Errorf("error no load balancers defined for cluster")
+	}
+
+	// This function will check if the Load Balancer has an ID set, use that to check Status for the Load Balancer's hostname. Or, if Name is non-empty, lookup the current Load Balancer status for hostname. Other cases result in errors.
+	getHostName := func(loadBalancer infrav1beta2.VPCLoadBalancerSpec) (*string, error) {
+		// Attempt to retrieve the Load Balancer's hostname from Status by ID.
+		if loadBalancer.ID != nil {
+			if lb, ok := s.NetworkStatus().LoadBalancers[*loadBalancer.ID]; ok {
+				return lb.Hostname, nil
+			}
+			// If the Load Balancer isn't in Status, assume it is not ready/reconciled.
+			return nil, nil
+		}
+
+		// NOTE(cjschaef): As a default Load Balancer isn't supported (when none are provided in Spec), this should result in an error.
+		// TODO(cjschaef): Likely would be beneficial to have webhook validation for this case.
+		if loadBalancer.Name == "" {
+			return nil, fmt.Errorf("error load balancer spec has no id or name defined")
+		}
+		// Lookup current Load Balancer hostname, via its Status, based on name, as ID was already found to be nil.
+		loadBalancerStatus, err := s.getLoadBalancer(loadBalancer)
+		if err != nil {
+			return nil, fmt.Errorf("error retrieving load balancer status: %w", err)
+		} else if loadBalancerStatus == nil {
+			// If no Load Balancer Status is found, return an error.
+			return nil, fmt.Errorf("error no load balancer status available for %s", loadBalancer.Name)
+		}
+		return loadBalancerStatus.Hostname, nil
+	}
+
+	// If only one Load Balancer was provided, attempt to use that Load Balancer's hostname.
+	if len(s.NetworkSpec().LoadBalancers) == 1 {
+		return getHostName(s.NetworkSpec().LoadBalancers[0])
```

Comment:
Any reason for handling this special case, In case of 1 lb defined it can be private as well is this the reason?

Reply:
Yes, if a Public LB was not found, and there was only one LB, we expect it must be a Private LB, and Public traffic isn't expected to be enabled. This isn't the normal path, but still need to cover this design.